community/server/src/main/java/org/neo4j/server/configuration/ConfigLoader.java

```diff
@@ -96,8 +96,10 @@ private static void overrideEmbeddedDefaults( HashMap<String, String> config )
         config.putIfAbsent( ShellSettings.remote_shell_enabled.name(), TRUE );
         config.putIfAbsent( GraphDatabaseSettings.logs_directory.name(), "logs" );
         config.putIfAbsent( GraphDatabaseSettings.auth_enabled.name(), "true" );
+
+        String dataDirectory = config.getOrDefault( data_directory.name(), data_directory.getDefaultValue() );
         config.putIfAbsent( GraphDatabaseSettings.auth_store.name(),
-                new File( config.get( data_directory.name() ), "dbms/auth" ).toString() );
+                new File( dataDirectory, "dbms/auth" ).toString() );
     }
```

Comment:
I am not following, do you mean first put the data directory in the map and then use that value to figure out where to put the auth store? The intention of this method as I have understood it is to override default values specified in an embedded installation, embedded installations does not have a notion of a data directory (which was as I understood it the primary reason why @benbc  made this change in the first place) so it feels somewhat backwards to override that value, and the default value should still be `data_directory.getDefaultValue()`, right?